src/components/product/migrations/fix-nan-total-verse-equivalents.migration.ts

```diff
@@ -0,0 +1,52 @@
+import { node, relation } from 'cypher-query-builder';
+import { ID } from '~/common';
+import { BaseMigration, Migration } from '~/core/database';
+import { ACTIVE } from '~/core/database/query';
+import { getTotalVerseEquivalents } from '../../scripture/verse-equivalents';
+import { ProductService } from '../product.service';
+
+@Migration('2023-10-27T09:47:07')
+export class FixNaNTotalVerseEquivalentsMigration extends BaseMigration {
+  constructor(private readonly productService: ProductService) {
+    super();
+  }
+
+  async up() {
+    const session = this.fakeAdminSession;
+    const ids = await this.db
+      .query()
+      .match([
+        node('product', 'Product'),
+        relation('out', '', 'totalVerseEquivalents', ACTIVE),
+        node('tve', 'Property'),
+      ])
+      .raw('WHERE tve.hadNaN = true or isNaN(tve.value)')
+      .return<{ id: ID }>('product.id as id')
+      .map('id')
+      .run();
+
+    const products = await this.productService.readManyUnsecured(
+      ids,
+      this.fakeAdminSession,
+    );
+
+    for (const p of products) {
+      const correctTotalVerseEquivalent = getTotalVerseEquivalents(
+        ...p.scriptureReferences,
+      );
+
+      if (p.__typename === 'DirectScriptureProduct') {
```

Comment:
Should anything be logged or done if the type is something other than one of these two?

Reply:
I was going to add a logging statement, but then I noticed that out of the 60 products 24 are DirectScriptureProducts and the remaining 36 are DerivativeScriptureProducts. Do you think I should add one anyways for "safety" sake?

Reply:
Nah. These are the only two types with this property.